README.md

```diff
@@ -2381,6 +2383,77 @@ Notice that the [`Browser`](#browser) is an immutable object, i.e. this
 method actually returns a *new* [`Browser`](#browser) instance with the
 given setting applied.
 
+
+#### withHeader()
+
+The `withHeader(string $header, string $value): Browser` method can be used to
+add another header.
+For example:
+
+```php
+$browser = $browser->withHeader('User-Agent', 'ACMC');
+
+$browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+    // case-insensitive
+    var_dump($response->getHeader('user-agent'));
+}, function (Exception $e) {
+    echo 'Error: ' . $e->getMessage() . PHP_EOL;
+});
+```
+
+Note that the new header will (case-insensitive) overwrite an existing header with the same name.
+Headers passed to the HTTP methods will be preferred.
+
+```php
+$browser = $browser->withHeader('User-Agent', 'ACMC');
+
+$browser->get($url, array('user-agent' => 'value'))->then(function (Psr\Http\Message\ResponseInterface $response) {
+    // case-insensitive
+    var_dump($response->getHeaders()); 
+    //will return array(
+    //                    ...
+    //                    'user-agent' => 'value'
+    //                    ...                                      
+    //                 )
+}, function (Exception $e) {
+    echo 'Error: ' . $e->getMessage() . PHP_EOL;
+});
+```
+
+
+#### withoutHeader()
+
+The `withoutHeader(string $header): Browser` method can be used to remove an added header
+(which got added via the `withHeader(string $header, string $value): Browser`).
+This works case-insensitive.
+It won't remove headers which will be passed directly into the HTTP method
+For example:
+
+```php
+$browser = $browser->withHeader('User-Agent', 'ACMC');
+$browser = $browser->withoutHeader('USER-AGENT');
+
+$browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+    // case-insensitive
+    var_dump($response->getHeader('User-Agent')); // will return empty array
```

Comment:
Same problem as above, this is dumping a response header which is unlikely to exist.